src/main/java/racingcar/domain/car/Car.java

```diff
@@ -0,0 +1,39 @@
+package racingcar.domain.car;
+
+public class Car {
+    private final Name name;
+    private final Distance distance;
+
+    private Car(Name name, Distance distance) {
+        this.name = name;
+        this.distance = distance;
+    }
+
+    public static Car of(String name) {
+        return new Car(
+                Name.of(name),
+                Distance.zero()
+        );
+    }
+
+    public Car move(Distance newDistance) {
+        return new Car(name, newDistance);
+    }
+
+    public boolean hasPosition(int distance) {
+        return Integer.valueOf(distance).equals(this.distance.getValue());
+    }
```

Comment:
```suggestion
    public boolean hasPosition(int distance) {
        return distance == this.getDistanceValue();
    }
```

비교 대상이 둘 다 Integer 가 아니라 int 니 `==` 로도 충분할 것 같아요😊
오히려 지금 상황에서 Object 가 제공하는 equals 를 사용하면 불필요한 boxing / unboxing 연산이 들어갈 수 있어요.!